pkg/controller/infrastructure/actuator.go

```diff
@@ -484,6 +484,15 @@ func (a *actuator) Delete(ctx context.Context, infra *extensionsv1alpha1.Infrast
 		return tf.CleanupConfiguration(ctx)
 	}
 
+	resIsEmpty, err := common.IsResourceEmpty(tf)
```

Comment:
I would combine lines 481-485 in this new `IsResourceEmpty` function somehow.

Reply:
good suggestion, I will combine these lines together